pkg/lib/bundle/validate.go

```diff
@@ -160,16 +191,35 @@ func (i imageValidator) ValidateBundleFormat(directory string) error {
 		}
 	}
 
-	_, err = ValidateChannelDefault(annotations[ChannelsLabel], annotations[ChannelDefaultLabel])
+	_, err := ValidateChannelDefault(annotations[ChannelsLabel], annotations[ChannelDefaultLabel])
 	if err != nil {
 		validationErrors = append(validationErrors, err)
 	}
+	return validationErrors
+}
 
-	if len(validationErrors) > 0 {
-		return NewValidationError(validationErrors)
+// Validate the dependencies file
+func validateDependencies(dependenciesFile *registry.DependenciesFile) []error {
+	var validationErrors []error
+
+	// Validate dependencies if exists
+	for _, d := range dependenciesFile.Dependencies {
+		dep := d.GetTypeValue()
+		errs := []error{}
+		if dep != nil {
+			switch d := dep.(type) {
+			case registry.GVKDependency:
+				errs = d.Validate()
+			case registry.PackageDependency:
+				errs = d.Validate()
+			}
```

Comment:
Does it make sense to add a default switch case that returns an error? I imagine that this is coupled with GetTypeValue and it would probably be a bug if GetTypeValue were extended without adding a corresponding case here.